scripts/artifacts-building/README.md

```diff
@@ -0,0 +1,115 @@
+# Artifacts documentation
+
+#### Overview
+
+RPC-O builds and uses the following artifacts:
+
+* apt: All the apt repositories used in RPC-O and OpenStack-Ansible (OSA) are
+  aggregated into a single apt repository using
+  [aptly](https://www.aptly.info/). The reason for doing this is to ensure
+  that every single deployment of RPC-O uses exactly the same apt packages
+  for each release.
+
+* git: All the git repositories which are synchronised by the repo container
+  are artifacted as compressed tarballs so that they are quicker and more
+  reliable to download.
+
+* python: All python wheels and venvs are built using the apt artifacts so
+  that these can be used across all RPC-O environments.
+
+* container: Using the apt and python artifacts, most roles implemented in
+  RPC-O have a container variant chroot prepared so that deployments can
+  consume them to ensure consistency across all RPC-O environments.
+
+Artifacts in RPC-O are both built and consumed using scripts in this
+directory.
+
+#### Artifact Build Process
+
+The CI process which builds RPC-O artifacts executes them in a specific
+order to ensure that each artifact is using the previous artifacts which
+were built.
+
+1. Apt artifacts are built. This is done per RPC-O series and published
+   using the value for ``rpc_release`` and the distribution code name.
+   Current apt repositories available can be found on
+   [rpc-repo](http://rpc-repo.rackspace.com/apt-mirror/integrated/dists/).
+
+   The upstream apt repositories to mirror are all set in
+   ``scripts/artifacts-building/apt/aptly-vars.yml`` in the ``aptly_mirrors``
+   variable. They are combined into snapshots per release, then combined into
+   an integrated snapshot as specified in the variable ``aptly_miko_mapping``.
+
+   Apt repositories which provide the same package name across two different
+   distributions with sdifferent contents have to be published as independent
+   repositories. Those are defined in the ``aptly_n_mapping`` variable.
+
+   Only one Apt artifact build can be executed at a time across all
+   series/tests as the test executes on a long running host which has
+   aptly installed. Aptly's CLI locks the database and can therefore only
+   execute one job at a time. The job executes
+   ``scripts/artifacts-building/apt/build-apt-artifacts.sh``.
+
+2. Git artifacts are built. This is per RPC-O series and published using
+   the value for ``rpc_release``. Published artifacts can be found on
+   [rpc-repo](http://rpc-repo.rackspace.com/git-archives/). The job
+   executes ``scripts/artifacts-building/git/build-git-artifacts.sh``.
+
```

Comment:
This uses OSA lookup py_pkgs to discover what to build. It therefore relies on path given to it, git clones and then prepare a .tgz and a sha1sum for it.

Reply:
That's detail which can be discovered by looking at the script itself. It's not a complicated script and I'd rather not provide so much detail here when the process may change in the future.